File: java-suite/src/main/java/de/fraunhofer/iem/classesAndObjects/enumClass/enumWithAbstractMethods/Menu.java
package de.fraunhofer.iem.classesAndObjects.enumClass.enumWithAbstractMethods;

/**
 * Demonstrates the Enum with abstract methods in Java
 *
 * @author Ranjith Krishnamurthy
 */
public enum Menu {
    CHICKEN_GRAVY(120) {
        @Override
        public void print() {
            System.out.println("Chicken gravy costs " + this.price);
        }
    },
    EGG_OMELET(20) {
        @Override
        public void print() {
            System.out.println("Egg omelet costs " + this.price);
        }
    },
    CHICKEN_BIRYANI(250) {
        @Override
        public void print() {
            System.out.println("Chicken biryani costs " + this.price);
        }
    },
    VEG_PULAO(90) {
        @Override
        public void print() {
            System.out.println("Veg pulao costs " + this.price);
        }
    };

    public int price;

    Menu(int price) {
        this.price = price;
    }

    public abstract void print();
}

File: kotlin-suite/src/main/kotlin/de/fraunhofer/iem/classesAndObjects/enumClass/anonymousClassInEnum/Menu.kt
package de.fraunhofer.iem.classesAndObjects.enumClass.anonymousClassInEnum

/**
 * Demonstrates the anonymous class in Enum in Kotlin
 *
 * @author Ranjith Krishnamurthy
 */
enum class Menu(val price: Int) {
    CHICKEN_GRAVY(120) {
        override fun print() {
            println("Chicken gravy costs $price")
        }
    },
    EGG_OMELET(20) {
        override fun print() {
            println("Egg omelet costs $price")
        }
    },
    CHICKEN_BIRYANI(250) {
        override fun print() {
            println("Chicken biryani costs $price")
        }
    },
    VEG_PULAO(90) {
        override fun print() {
            println("Veg pulao costs $price")
        }
    };

    abstract fun print()
}